src/kdc/kdc_preauth.c

```diff
@@ -888,17 +885,24 @@ get_preauth_hint_list(krb5_kdc_req *request, krb5_kdcpreauth_rock rock,
     state->rock = rock;
     state->realm = rock->rstate->realm_data;
     state->e_data_out = e_data_out;
+    state->pa_data = NULL;
+    state->ap = preauth_systems;
 
-    state->pa_data = calloc(n_preauth_systems + 1, sizeof(krb5_pa_data *));
-    if (!state->pa_data) {
-        free(state);
-        (*respond)(arg);
-        return;
-    }
+    if (alloc_pa_data(KRB5_PADATA_FX_FAST, 0, &pa) != 0)
```

Comment:
Nit: these should be using an intermediate variable (e.g., `ret`).

Reply:
If we were going to do anything with the result besides test it for success, I would agree, but get_preauth_hint_list() does nothing on failure besides stop (this dates back to it being a function returning void in 1.0 and is certainly questionable).